packages/protocol/contracts-0.8/common/EpochManager.sol

```diff
@@ -400,6 +402,38 @@ contract EpochManager is
     return epochs[epoch].lastBlock;
   }
 
+  /**
+   * @notice Returns the epoch number of a specified blockNumber.
+   * @param _blockNumber Block number of the epoch info is retreived.
+   */
+  function getEpochNumberOfBlock(uint256 _blockNumber) external view returns (uint256) {
+    (uint256 _epochNumber, , , , , ) = _getEpochByBlockNumber(_blockNumber);
```

Comment:
Is there a way to do this without calling `_getEpochByBlockNumber`? If this is expected to be used in transactions rather than only off-chain view calls (by us or any third party protocol), reading the entire `elected` array into memory will waste a good bit of gas when only the epoch number is needed.